cmd/data-service/service/task/management.go

```diff
@@ -0,0 +1,326 @@
+	}
+
+	return nil, nil
+}
+
+// ListTaskManagement list task management.
+func (svc *service) ListTaskManagement(cts *rest.Contexts) (interface{}, error) {
+	req := new(core.ListReq)
+	if err := cts.DecodeInto(req); err != nil {
+		return nil, err
+	}
+
+	if err := req.Validate(); err != nil {
+		return nil, errf.NewFromErr(errf.InvalidParameter, err)
+	}
+
+	opt := &types.ListOption{
+		Filter: req.Filter,
+		Page:   req.Page,
+		Fields: req.Fields,
+	}
+	res, err := svc.dao.TaskManagement().List(cts.Kit, opt)
+	if err != nil {
+		logs.Errorf("list task management failed, err: %v, rid: %s", err, cts.Kit.Rid)
+		return nil, fmt.Errorf("list task management failed, err: %v", err)
+	}
+	if req.Page.Count {
+		return &task.ListManagementResult{Count: res.Count}, nil
+	}
+
+	managements := make([]coretask.Management, 0, len(res.Details))
+	for _, one := range res.Details {
+		extension := new(coretask.ManagementExt)
+		if len(one.Extension) != 0 {
+			if err = json.UnmarshalFromString(string(one.Extension), &extension); err != nil {
+				return nil, fmt.Errorf("UnmarshalFromString json extension failed, err: %v", err)
+			}
+		}
+		operations := make([]enumor.TaskOperation, len(one.Operations))
+		for i, operation := range one.Operations {
+			operations[i] = enumor.TaskOperation(operation)
+		}
+
+		managements = append(managements, coretask.Management{
+			ID:         one.ID,
+			BkBizID:    one.BkBizID,
+			Source:     one.Source,
+			Vendor:     one.Vendor,
+			State:      one.State,
+			AccountID:  one.AccountID,
+			Resource:   one.Resource,
+			Operations: operations,
+			FlowIDs:    one.FlowIDs,
+			Extension:  extension,
+			Revision: core.Revision{
+				Creator:   one.Creator,
+				Reviser:   one.Reviser,
+				CreatedAt: one.CreatedAt.String(),
+				UpdatedAt: one.UpdatedAt.String(),
+			},
+		})
+	}
+
+	return &task.ListManagementResult{Managements: managements}, nil
+}
+
+// CancelTaskManagement cancel task management.
+func (svc *service) CancelTaskManagement(cts *rest.Contexts) (interface{}, error) {
+	req := new(task.CancelReq)
+	if err := cts.DecodeInto(req); err != nil {
+		return nil, errf.NewFromErr(errf.DecodeRequestFailed, err)
+	}
+
+	if err := req.Validate(); err != nil {
+		return nil, errf.NewFromErr(errf.InvalidParameter, err)
+	}
+
+	_, err := svc.dao.Txn().AutoTxn(cts.Kit, func(txn *sqlx.Tx, opt *orm.TxnOption) (interface{}, error) {
+		managementFilter := tools.ContainersExpression("id", req.IDs)
+		managementUpdate := &tabletask.ManagementTable{State: enumor.TaskManagementCancel}
+		if err := svc.dao.TaskManagement().UpdateWithTx(cts.Kit, txn, managementFilter, managementUpdate); err != nil {
+			logs.Errorf("update task management failed, err: %v, req: %+v, rid: %s", err, req, cts.Kit.Rid)
+			return nil, fmt.Errorf("cancel task management failed, err: %v", err)
+		}
+
+		taskFilter, err := tools.And(tools.ContainersExpression("task_management_id", req.IDs),
+			tools.EqualExpression("state", enumor.TaskDetailInit))
```

Comment:
用ExpressionXXX 和 RuleXXX可以打平filter降低嵌套层级
```suggestion
		taskFilter, err := tools.ExpressionAnd(tools.RuleIn("task_management_id", req.IDs),
			tools.RuleEqual("state", enumor.TaskDetailInit))
```

Reply:
已调整